platform/commons/src/main/java/com/peregrine/commons/util/PerUtil.java

```diff
@@ -914,4 +915,68 @@ public boolean doAdd(final Resource resource) {
         @Override
         public boolean doAddChildren(Resource resource) { return true; }
     }
+
+    /**
+     * Extracts the tenant name from the resource
+     * @param resource a resource
+     * @return tenant name on success, and <code>null</code> otherwise
+     */
+    public static String getTenantNameFromResource(final Resource resource) {
+        return (resource != null)
+                ? getTenantNameFromPath(resource.getPath())
+                : null;
+    }
+
+    /**
+     * Extracts the tenant name for the path
+     * @param path a path
+     * @return tenant name on success, and <code>null</code> otherwise
+     */
+    public static String getTenantNameFromPath(final String path) {
```

Comment:
make sure `/content/iamatenant` also works - looking at the pattern this will not work right now

Reply:
I tested this use case on https://www.regexplanet.com/advanced/java/index.html. It works.

<img width="1678" alt="Screen Shot 2020-08-25 at 8 55 52 AM" src="https://user-images.githubusercontent.com/1387966/91198054-ecd16580-e6b0-11ea-9b19-2fe4e3a1dec3.png">